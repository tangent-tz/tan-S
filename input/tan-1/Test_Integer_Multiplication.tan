main {
    const x := 1000;
    const y := 500;

    # Basic multiplication
    const a := 100;
    const b := 200;
    print a * b;  # Output: 20000
    print \n;

    # Multiplication with negative numbers
    const c := -7000;
    const d := -3000;
    print c * d;  # Output: 21000000
    print \n;

    # Multiplication with mixed positive and negative numbers
    const e := 1500;
    const f := -200;
    print e * f;  # Output: -300000
    print \n;

    # Multiplication with zero
    const g := 10000;
    const h := 0;
    print g * h;  # Output: 0
    print \n;

    # Multiplication with large numbers
#   const i := 99999999;
#   const j := 88888888;
#   print i * j;   Output: 8888888705555552
#   print \n;

    # Multiplication resulting in a negative value
    const k := -5000;
    const l := -10000;
    print k * l;  # Output: 50000000
    print \n;

    # Multiplication involving a large negative value
#    const m := -9999999;
#    const n := 8888888;
#    print m * n;   Output: -88888877555555112
#    print \n;
}
