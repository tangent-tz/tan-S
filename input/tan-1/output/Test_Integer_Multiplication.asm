        Jump         $$main                    
        DLabel       $eat-location-zero        
        DataZ        8                         
        DLabel       $print-format-integer     
        DataC        37                        %% "%d"
        DataC        100                       
        DataC        0                         
        DLabel       $print-format-boolean     
        DataC        37                        %% "%s"
        DataC        115                       
        DataC        0                         
        DLabel       $print-format-float       
        DataC        37                        %% "%f"
        DataC        102                       
        DataC        0                         
        DLabel       $print-format-character   
        DataC        37                        %% "%c"
        DataC        99                        
        DataC        0                         
        DLabel       $print-format-string      
        DataC        37                        %% "%s"
        DataC        115                       
        DataC        0                         
        DLabel       $print-format-newline     
        DataC        10                        %% "\n"
        DataC        0                         
        DLabel       $print-format-space       
        DataC        32                        %% " "
        DataC        0                         
        DLabel       $print-format-tab         
        DataC        9                         %% "\t"
        DataC        0                         
        DLabel       $boolean-true-string      
        DataC        116                       %% "true"
        DataC        114                       
        DataC        117                       
        DataC        101                       
        DataC        0                         
        DLabel       $boolean-false-string     
        DataC        102                       %% "false"
        DataC        97                        
        DataC        108                       
        DataC        115                       
        DataC        101                       
        DataC        0                         
        DLabel       $errors-general-message   
        DataC        82                        %% "Runtime error: %s\n"
        DataC        117                       
        DataC        110                       
        DataC        116                       
        DataC        105                       
        DataC        109                       
        DataC        101                       
        DataC        32                        
        DataC        101                       
        DataC        114                       
        DataC        114                       
        DataC        111                       
        DataC        114                       
        DataC        58                        
        DataC        32                        
        DataC        37                        
        DataC        115                       
        DataC        10                        
        DataC        0                         
        Label        $$general-runtime-error   
        PushD        $errors-general-message   
        Printf                                 
        Halt                                   
        DLabel       $errors-int-divide-by-zero 
        DataC        105                       %% "integer divide by zero"
        DataC        110                       
        DataC        116                       
        DataC        101                       
        DataC        103                       
        DataC        101                       
        DataC        114                       
        DataC        32                        
        DataC        100                       
        DataC        105                       
        DataC        118                       
        DataC        105                       
        DataC        100                       
        DataC        101                       
        DataC        32                        
        DataC        98                        
        DataC        121                       
        DataC        32                        
        DataC        122                       
        DataC        101                       
        DataC        114                       
        DataC        111                       
        DataC        0                         
        Label        $$i-divide-by-zero        
        PushD        $errors-int-divide-by-zero 
        Jump         $$general-runtime-error   
        DLabel       $errors-float-divide-by-zero 
        DataC        102                       %% "float divide by zero"
        DataC        108                       
        DataC        111                       
        DataC        97                        
        DataC        116                       
        DataC        32                        
        DataC        100                       
        DataC        105                       
        DataC        118                       
        DataC        105                       
        DataC        100                       
        DataC        101                       
        DataC        32                        
        DataC        98                        
        DataC        121                       
        DataC        32                        
        DataC        122                       
        DataC        101                       
        DataC        114                       
        DataC        111                       
        DataC        0                         
        Label        $$f-divide-by-zero        
        PushD        $errors-float-divide-by-zero 
        Jump         $$general-runtime-error   
        DLabel       $usable-memory-start      
        DLabel       $global-memory-block      
        DataZ        48                        
        Label        $$main                    
        PushD        $global-memory-block      
        PushI        0                         
        Add                                    %% x
        PushI        1000                      
        StoreI                                 
        PushD        $global-memory-block      
        PushI        4                         
        Add                                    %% y
        PushI        500                       
        StoreI                                 
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% a
        PushI        100                       
        StoreI                                 
        PushD        $global-memory-block      
        PushI        12                        
        Add                                    %% b
        PushI        200                       
        StoreI                                 
        PushD        $global-memory-block      
        PushI        8                         
        Add                                    %% a
        LoadI                                  
        PushD        $global-memory-block      
        PushI        12                        
        Add                                    %% b
        LoadI                                  
        Multiply                               
        PushD        $print-format-integer     
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        16                        
        Add                                    %% c
        PushI        7000                      
        Negate                                 
        StoreI                                 
        PushD        $global-memory-block      
        PushI        20                        
        Add                                    %% d
        PushI        3000                      
        Negate                                 
        StoreI                                 
        PushD        $global-memory-block      
        PushI        16                        
        Add                                    %% c
        LoadI                                  
        PushD        $global-memory-block      
        PushI        20                        
        Add                                    %% d
        LoadI                                  
        Multiply                               
        PushD        $print-format-integer     
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        24                        
        Add                                    %% e
        PushI        1500                      
        StoreI                                 
        PushD        $global-memory-block      
        PushI        28                        
        Add                                    %% f
        PushI        200                       
        Negate                                 
        StoreI                                 
        PushD        $global-memory-block      
        PushI        24                        
        Add                                    %% e
        LoadI                                  
        PushD        $global-memory-block      
        PushI        28                        
        Add                                    %% f
        LoadI                                  
        Multiply                               
        PushD        $print-format-integer     
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        32                        
        Add                                    %% g
        PushI        10000                     
        StoreI                                 
        PushD        $global-memory-block      
        PushI        36                        
        Add                                    %% h
        PushI        0                         
        StoreI                                 
        PushD        $global-memory-block      
        PushI        32                        
        Add                                    %% g
        LoadI                                  
        PushD        $global-memory-block      
        PushI        36                        
        Add                                    %% h
        LoadI                                  
        Multiply                               
        PushD        $print-format-integer     
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        PushD        $global-memory-block      
        PushI        40                        
        Add                                    %% k
        PushI        5000                      
        Negate                                 
        StoreI                                 
        PushD        $global-memory-block      
        PushI        44                        
        Add                                    %% l
        PushI        10000                     
        Negate                                 
        StoreI                                 
        PushD        $global-memory-block      
        PushI        40                        
        Add                                    %% k
        LoadI                                  
        PushD        $global-memory-block      
        PushI        44                        
        Add                                    %% l
        LoadI                                  
        Multiply                               
        PushD        $print-format-integer     
        Printf                                 
        PushD        $print-format-newline     
        Printf                                 
        Halt                                   
