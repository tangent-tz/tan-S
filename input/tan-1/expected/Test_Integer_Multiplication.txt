20000
21000000
-300000
0
50000000
